### ISSUE TITLE
max-complexity option is unrecognized

### ISSUE THREAD

**Issue by Coacher** [PyCQA/pylint#10998]
### Bug description

```python
Docs mention --max-complexity CLI argument. Docs also say --max-complexity should be under [tool.pylint.design] section in pyproject.toml: https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html#max-complexity.

However, setting this option either as a CLI argument or via pyproject.toml results in unrecognized option error.
```

### Configuration

```ini
$ cat pyproject.toml
[tool.pylint.design]
max-complexity = 20
```

### Command used

```shell
$ >empty.py
$ pylint empty.py
# OR remove pyproject.toml and use the option directly
$ pylint --max-complexity=20 empty.py
```

### Pylint output

```python
************* Module /home/user/.virtualenvs/tmp-77974fdde2cbc4c/pyproject.toml
pyproject.toml:1:0: E0015: Unrecognized option found: max-complexity (unrecognized-option)
```

### Expected behavior

max-complexity option is recognized.

### Pylint version

```shell
pylint 4.0.5
astroid 4.0.4
Python 3.12.3 (main, Mar 23 2026, 19:04:32) [GCC 13.3.0]
```

### OS / Environment

Ubuntu 24.04.4

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
You'll also need to enable the `pylint.extensions.mccabe` plugin for this to become an available option :)
